### PR TITLE
Change default library installation path and check permissions before build

### PR DESCRIPTION
**Pull request by skypodolsky**
1. No generic way to locate the user library paths was found for all distros we support
2. Properly, this should have been solved with Autoconf, which we don't use at the moment
3. But as we produce deb/rpm packages, this should not be a problem for users. Still, for developers, there's a possibility to specify the installation prefix with `sudo make PREFIX=<prefix>`
5. Apart from that, there should always be a common path for the libraries: /usr/lib64
6. Hence, I've replaced the prefix to `/usr`, making the default installation path of `/usr/lib64`
7. Apart from that, added permissions check to make sure that driver is compiled with root rights so that kernel functions are located properly in `kernel-configs.h`

### CODE REVIEW

**Review by vsazhenyuk-softheme — APPROVED** [review]

**Review by anelson — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ export CC = gcc
 export RM = rm -f
 CFLAGS ?= -Wall
 export CCFLAGS = $(CFLAGS) -std=gnu99
-export PREFIX = /usr/local
+export PREFIX = /usr
 export BASE_DIR = $(abspath .)
 
 BUILDDIR := $(CURDIR)/pkgbuild
```
Comment (e-kov) [inline Makefile:10]:
```suggestion
EUID := $(shell id -u -r)

BUILDDIR := $(CURDIR)/pkgbuild
```

File: Makefile [hunk 2]
```diff
@@ -16,9 +16,12 @@ PKGBUILDFLAGS := --define "_topdir $(BUILDDIR)" -ba --with devmode
...
 			$(BUILDDIR)/SOURCES $(BUILDDIR)/SPECS $(BUILDDIR)/BUILD $(BUILDDIR)/BUILDROOT
 
-.PHONY: all driver library-shared library-static library application application-shared utils clean install uninstall pkgclean pkgprep deb rpm
+.PHONY: all check_root driver library-shared library-static library application application-shared utils clean install uninstall pkgclean pkgprep deb rpm
 
-all: driver library application utils
+all: check_root driver library application utils
+
+check_root:
+	@if ! [ "$(shell id -u)" = 0 ];then echo "You are not root, run this target as root, please"; exit 1; fi
```
Comment (e-kov) [inline Makefile:24]:
```suggestion
ifneq ($(EUID),0)
	@echo "Run as sudo or root."
	@exit 1
endif
```

File: Makefile [hunk 3]
```diff
@@ -16,9 +18,15 @@ PKGBUILDFLAGS := --define "_topdir $(BUILDDIR)" -ba --with devmode
...
+all: check_root driver library application utils
 
-all: driver library application utils
+check_root:
+ifneq ($(EUID),0)
+	@echo "Run as sudo or root."
+	@exit 1
+endif
 
 driver:
```
Comment (e-kov) [inline Makefile:31]: One small detail )
```suggestion
driver: check_root
```

**Review by e-kov — APPROVED** [review]
Now this change looks good :+1:

**Comment by e-kov** [comment]
@skypodolsky please close #115
You forgot to write `closes #115` in the PR's description. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Pull request merged**